gapis/api/vulkan/frame_loop.go

```diff
@@ -779,14 +791,75 @@ func (f *frameLoop) buildStartEndStates(ctx context.Context, startState *api.Glo
 				f.renderPassToCreate[renderPass] = true
 			}
 
-			// TODO:  Recreate destroyed resources.
-		}
+		// Command Buffers
+		case *VkAllocateCommandBuffers:
+			vkCmd := cmd.(*VkAllocateCommandBuffers)
+			cmdBuffers := vkCmd.PCommandBuffers().MustRead(ctx, vkCmd, currentState, nil)
+			f.commandBufferToFree[cmdBuffers] = true
+			log.D(ctx, "Command buffer %v allcoated.", cmdBuffers)
+		case *VkFreeCommandBuffers:
+			vkCmd := cmd.(*VkFreeCommandBuffers)
+			cmdBufCount := vkCmd.CommandBufferCount()
+			cmdBufs := vkCmd.PCommandBuffers().Slice(0, uint64(cmdBufCount), currentState.MemoryLayout).MustRead(ctx, cmd, currentState, nil)
+			for _, cmdBuf := range cmdBufs {
+				// The command buffer deleted in this call was created during loop, no action needed.
+				log.D(ctx, "Command buffer %v freed", cmdBufs)
+				if _, ok := f.commandBufferToFree[cmdBuf]; ok {
+					delete(f.commandBufferToFree, cmdBuf)
+				} else { // The command buffer deleted in this call was not created during loop, need to back up it
+					f.commandBufferToAllocate[cmdBuf] = true
+				}
+			}
+		case *VkBeginCommandBuffer:
+			vkCmd := cmd.(*VkBeginCommandBuffer)
+			cmdBuf := vkCmd.CommandBuffer()
+			f.commandBufferRecorded[cmdBuf] = true
+			log.D(ctx, "Command buffer %v began", cmdBuf)
+
+			// If this command buffer is allocated during loop then it is reset by the re-allocation step.
+			if _, ok := f.commandBufferToFree[cmdBuf]; ok {
+				break
+			}
+			// If this command buffer has not been submitted before no action needed.
+			if _, ok := f.commandBufferNotRecorded[cmdBuf]; !ok {
```

Comment:
What if it was recorded just before the loop began?

Reply:
so here we're checking this condition: 
If it was recorded before the start of the loop:
a), if it was submitted during the loop and then at this point start to recording again, then we need to mark is as changed.
b), if it was not submitted before this point, we don't need to to anything